ruby/ql/test/library-tests/dataflow/array-flow/array_flow.rb

```diff
@@ -0,0 +1,523 @@
+        sink(x[0]) # $ hasValueFlow=25
+    end
+end
+
+def m26
+    a = [0, 1, source(26)]
+    b = a.compact
+    sink(b[0]) # $ hasValueFlow=26
+end
+
+def m27
+    a = [0, 1, source(27.1)]
+    b = [0, 1, source(27.2)]
+    a.concat(b)
+    sink(a[0]) # $ hasValueFlow=27.2
+    sink(a[2]) # $ hasValueFlow=27.1 $ hasValueFlow=27.2
+end
+
+def m28
+    a = [0, 1, source(28)]
+    a.count do |x|
+        sink x # $ hasValueFlow=28
+    end
+end
+
+def m29
+    a = [0, 1, source(29)]
+    a.cycle(2) do |x|
+        sink x # $ hasValueFlow=29
+    end
+end
+
+def m30
+    a = [0, 1, source(30.1)]
+    b = a.delete(2) { source(30.2) }
+    sink b # $ hasValueFlow=30.1 $ hasValueFlow=30.2
+end
+
+def m31
+    a = [0, 1, source(31)]
+    b = a.delete_at(2)
+    sink b # $ hasValueFlow=31
+end
+
+def m32
+    a = [0, 1, source(32)]
+    b = a.delete_if do |x|
+        sink x # $ hasValueFlow=32
+    end
+    sink(b[0]) # $ hasValueFlow=32
+end
+
+def m33
+    a = [0, 1, source(33)]
+    b = a.difference([1])
+    sink(b[0]) # $ hasValueFlow=33
+end
+
+def m34(i)
+    a = [0, 1, source(34.1), [0, source(34.2)]]
+    sink(a.dig(0))
+    sink(a.dig(2)) # $ hasValueFlow=34.1
+    sink(a.dig(i)) # $ hasValueFlow=34.1
+    sink(a.dig(3,0))
+    sink(a.dig(3,1)) # $ hasValueFlow=34.2
+end
+
+def m35
+    a = [0, 1, source(35.1)]
+    b = a.detect(-> { source(35.2) }) do |x|
+        sink x # $ hasValueFlow=35.1
+    end
+    sink b # $ hasValueFlow=35.1 $ hasValueFlow=35.2
+end
+
+def m36(i)
+    a = [0, 1, source(36.1), source(36.2)]
+    b = a.drop(i)
+    sink(b[0]) # $ hasValueFlow=36.1 # $ hasValueFlow=36.2
+    b = a.drop(1)
+    sink(b[0])
+    sink(b[1]) # $ hasValueFlow=36.1
+    sink(b[i]) # $ hasValueFlow=36.1 # $ hasValueFlow=36.2
+    a[i] = source(36.3)
+    b = a.drop(1)
+    sink(b[1]) # $ hasValueFlow=36.1 # $ hasValueFlow=36.3
+    c = b.drop(100)
+    sink(c[1]) # $ hasValueFlow=36.3
+end
+
+def m37
+    a = [0, 1, source(37.1), source(37.2)]
+    b = a.drop_while do |x|
+        sink x # $ hasValueFlow=37.1 # $ hasValueFlow=37.2
+    end
+    sink(b[0]) # $ hasValueFlow=37.1 # $ hasValueFlow=37.2
+end
+
+def m38
+    a = [0, 1, source(38)]
+    b = a.each do |x|
+        sink x # $ hasValueFlow=38
+    end
+    sink(b[2]) # $ hasValueFlow=38
+end
+
+def m39
+    a = [0, 1, source(39)]
+    b = for x in a # desugars to an `each` call
+        sink x # $ hasValueFlow=39
+    end
+    sink x # $ hasValueFlow=39
+    sink(b[2]) # $ hasValueFlow=39
+end
+
+def m40
+    a = [0, 1, source(40)]
+    a.each_cons(2) do |x|
+        sink (x[0]) # $ hasValueFlow=40
+    end
+end
+
+def m41
+    a = [0, 1, source(41)]
+    b = a.each_entry do |x|
+        sink x # $ hasValueFlow=41
+    end
+    sink(b[2]) # $ hasValueFlow=41
+end
+
+def m42
+    a = [0, 1, source(42)]
+    b = a.each_index do |x|
+        sink x
+    end
+    sink(b[2]) # $ hasValueFlow=42
+end
+
+def m43
+    a = [0, 1, 2, source(43)]
+    b = a.each_slice do |x|
```

Comment:
`Array#each_slice` takes one argument, the size of the slice that you want. So this should probably be
```rb
b = a.each_slice(1) do |x|
```